### PR TITLE
feat(mixpanel): track overlays usage

### PR DESCRIPTION
**Pull request by smbea**
Related to #2934

### CODE REVIEW

**Comment by marstamm** [comment]
Looks okay code-wise. Is there an easy way to test it?

**Comment by smbea** [comment]
> Looks okay code-wise. Is there an easy way to test it?

To test see the results in Mixpanel, you need to request access to the app in IT. You can also just add a `debug` option in the code so that it logs all the events being fired. Other than that, you can use the flags MIXPANEL_TOKEN (please reach out via slack) and MIXPANEL_STAGE  (="int").

File: client/src/plugins/user-journey-statistics/event-handlers/__tests__/OverlayEventHandlerSpec.js [hunk 1]
```diff
@@ -0,0 +1,448 @@
...
+          });
+        });
+
+      });
+
+
+    });
+
+
+    describe('should send engine profile', () => {
```
Comment (MaxTru) [inline client/src/plugins/user-journey-statistics/event-handlers/__tests__/OverlayEventHandlerSpec.js:254]: Could we test here that this profile is actually sent (just as in the other tests)? E.g.,
```
      expect(track).to.have.been.calledWith('overlay:versionInfo:opened', {
        executionPlatform: 'Camunda Platform'
      });
```

Or is there any reason not to do that?

**Review by MaxTru — CHANGES_REQUESTED** [review]

File: client/src/plugins/user-journey-statistics/event-handlers/__tests__/OverlayEventHandlerSpec.js (hunk 1 above)
Comment (MaxTru) [reply client/src/plugins/user-journey-statistics/event-handlers/__tests__/OverlayEventHandlerSpec.js:446]: I saw it diffing from the other test cases. But you are right, for `executionPlatform` we always test using this pattern (also in existing other test cases).

**Review by MaxTru — APPROVED** [review]
Cool.

LGTM

**Pull request merged**